Automattic/woocommerce-payments
# E2E subscription tests - Fix timeouts & optimize flow

Fixes #4404 
Fixes #4408

#### Changes proposed in this Pull Request

This PR mostly addresses flakiness present on subscription flows. Here are the major changes on this PR:

* Updates `woocommerce/e2e-utils` package from `0.1.6` to `0.2.0`.
* Adds new user account with the email `e2e-wcpay-subscriptions-customer@woocommerce.com` for running subscription tests. This prevents deleting the default customer account and causing problems with other test flows.
* Removes creating subscription products during tests. Instead imports the subscription products during E2E setup using an XML file.
* Updates flow to explicitly logout customer account after tests to prevent 500 internal server error while trying to login as admin and to prevent cascade failures.
* Deletes the flow `subscriptions/shopper/shopper-subscriptions-change-payment-method.spec.js` since the tests are already present on `subscriptions/shopper/shopper-subscriptions-manage-payments.spec.js`.
* Deletes the flow `subscriptions/shopper/shopper-subscriptions-purchase-subscription-product.spec.js` since it's covered in other tests
* Renamed `subscriptions/shopper/shopper-subscriptions-free-trial.spec.js` to `subscriptions/shopper/shopper-subscriptions-purchase-free-trial.spec.js`
* Moved the following tests from `subscriptions - merchant` to `subscriptions - shopper`
  * `subscriptions/merchant/merchant-subscriptions-purchase-no-sign-up.spec.js` to `subscriptions/shopper/shopper-subscriptions-purchase-no-sign-up-fee.spec.js` 
  * `subscriptions/merchant/merchant-subscriptions-purchase-sign-up-fee.spec.js` to `subscriptions/shopper/shopper-subscriptions-purchase-sign-up-fee.spec.js`
* Updates to subscription flows to fix test flakiness 


#### Testing instructions

* On GitHub actions, trigger a new E2E test against this branch and confirm that tests against WC `latest` & `beta` are passing. Tests against other versions are failing due to other reasons which will addressed in a different PR
* On local, setup a new E2E environment with local server instance & run `E2E_GROUP='subscriptions' E2E_BRANCH='merchant' npm run test:e2e` & `E2E_GROUP='subscriptions' E2E_BRANCH='shopper' npm run test:e2e` and confirm that the tests are passing.

-------------------


**Post merge**
